src/main/java/com/aol/cyclops2/data/collections/extensions/IndexedSequenceX.java

```diff
@@ -101,8 +104,21 @@
     @Override
     <R> IndexedSequenceX<R> flatMap(Function<? super T, ? extends Iterable<? extends R>> mapper);
 
-    
-    
-    
-    
+    /**
+     * Intercalate
+     *
+     * @param listOfLists Index of our sub sequence (inclusive)
+     * @return List with current IndexedSequenceX inserted between each List.
+     */
+    default List intercalate(List<List> listOfLists) {
```

Comment:
Can we make the return type

```java
ListX<R>
```


The parameters could be

```java
List<List<R>> listOfLists;
```

I would check that the parameter works for both List and classes that implement or extend it

```java
List<List<Integer>> listOfLists;
ListX<ListX<Integer>> listOfExtendedLists;
```